app/logic/daily_digest.rb

```diff
@@ -0,0 +1,32 @@
+class DailyDigest
+
+  def send
+    params = get_statistic
+    daily_statistic = DailyProgressDigest.new params
+
+    Notifier.digest(daily_statistic).deliver
+  end
+
+  private
+
+  def get_statistic
+    recipients = get_recipients
+    completed = completed_amount
+    remaining = incompleted_amount
+
+    Hash(recipients: recipients, completed: completed, remaining: remaining)
+  end
+
+  def get_recipients
+    User.all.map {|user| user.email}
+  end
```

Comment:
@vlasiak, Overlooked this one previously. There is a suggestion to use [pluck](http://apidock.com/rails/ActiveRecord/Calculations/pluck) in order to get needed attributes right for DB, instead of fetching objects into the memory.